adr/keycloak-adr.md

```diff
@@ -1,264 +1,139 @@
-
-#### nats.io references
-- [NATS Documentation](https://docs.nats.io/)
-- [NATS on Kubernetes](https://docs.nats.io/nats-on-kubernetes/nats-on-kubernetes)
-- [NATS Security](https://docs.nats.io/nats-on-kubernetes/nats-on-kubernetes)
+#### SAML Integration
+**Pros:**
+- Lets you automatically sync user groups between Keycloak and GitLab, making access control more accessible.
+
+**Cons:**
+- Setting it up can be complicated, so careful alignment between Keycloak and GitLab is needed.
+- To use all features, you might need a higher tier of GitLab, which could mean higher costs.
+- If you have a lot of users, you might need to pay for more expensive versions of Keycloak, which could be a bit of a hassle.
 
+### Provision Users
+Keycloak will provision users and manage user roles, groups, and permissions. This ensures user access is managed centrally and consistently across all applications and services.
 
+#### SCIM Integration
+**Pros:**
+- Easy to Mix Systems: SCIM is a common framework for sharing user info between different programs or online services, making them work smoothly.
+- Saves Time: It handles adding, updating, and removing user accounts by itself, which means less work for people managing these tasks.
+- Fewer Mistakes: Since it's automated, there's less chance someone will type something wrong or forget to update important info.
+
+**Cons:**
+- Can Be Tricky to Set Up: Getting SCIM to work just right with all your systems can be complicated.
+- Learning Curve: If you're not familiar with SCIM, it might take a while to get the hang of it.
+
+#### Manual Provisioning
+**Pros:**
+- Full control over the user provisioning process.
+- Can be implemented without additional development or integration efforts.
+
+**Cons:**
+- Time-consuming and not scalable.
+- Prone to human error, potentially leading to security and compliance issues.
+
+#### Custom Code
+**Pros:**
+- Flexibility: You can tailor the provisioning process to your exact needs, ensuring it fits your organization perfectly.
+- No exteranl depdency: You can do this with what GitLab already offers, no need for external OSS tools.
+
+**Cons:**
+- More Work: You'll have to write and maintain the code yourself, which can be a lot of work.
+
+### Deprovision Users
+Keycloak will be used to de-provision users and manage user access rights, ensuring timely revocation of access when users leave the organization or change roles.
+
+#### Automated Deprovisioning via SCIM
+**Pros:**
+- Ensures Timely Revocation of Access: Automatically removes users' access rights as soon as their status changes, improving security.
+- Reduces Manual Workload: Lessens the burden on IT staff by handling the de-provisioning process automatically, minimizing the chance of oversight.
+Supports Real-Time Updates: It can integrate with broader system architectures to ensure updates are processed immediately, keeping user access rights current.
```

Comment:
Add missing bullet point `-`
```suggestion
- Supports Real-Time Updates: It can integrate with broader system architectures to ensure updates are processed immediately, keeping user access rights current.
```